src/test/kotlin/ua/marchenko/artauction/user/repository/UserRepositoryTest.kt

```diff
@@ -68,43 +75,48 @@ class UserRepositoryTest : AbstractBaseIntegrationTest {
         val result = userRepository.existsByEmail(getRandomEmail())
 
         // THEN
-        assertFalse(result, "User with given email must not exist")
+        result.test()
+            .expectNext(false)
+            .`as`("User with given email must not exist")
+            .verifyComplete()
     }
 
     @Test
     fun `should return user when user with given email exists`() {
         // GIVEN
-        val savedUser = userRepository.save(MongoUser.random(id = null))
+        val savedUser = userRepository.save(MongoUser.random(id = null)).block()
 
         // WHEN
-        val foundUser = userRepository.findByEmail(savedUser.email!!)
+        val foundUser = userRepository.findByEmail(savedUser?.email!!)
 
         // THEN
-        assertEquals(savedUser.email, foundUser?.email)
+        foundUser.test()
+            .expectNext(savedUser)
+            .verifyComplete()
     }
 
     @Test
-    fun `should return null when user with given email doesnt exists`() {
+    fun `should return Mono empty() when user with given email doesnt exists`() {
         // WHEN
         val result = userRepository.findByEmail(getRandomEmail())
 
         // THEN
-        assertNull(result, "Found user must be null")
+        result.test()
+            .verifyComplete()
     }
 
     @Test
     fun `should return all users when they are exists`() {
         // GIVEN
-        val users = listOf(MongoUser.random(id = null))
-        users.forEach { user -> userRepository.save(user) }
+        val users = listOf(userRepository.save(MongoUser.random(id = null)).block())
 
         // WHEN
-        val result = userRepository.findAll()
+        val result = userRepository.findAll().collectList()
 
         // THEN
-        assertTrue(result.size >= users.size, "Size of list must be at least ${users.size}")
-        users.forEach { user ->
-            assertTrue(result.any { it.email == user.email }, "User with email ${user.email} must be found")
-        }
+        result.test()
+            .expectNextMatches { it.containsAll(users) }
+            .`as`("User with id ${users[0]?.id} must be found")
```

Comment:
What if we would extend this case with 2 users in list?

Reply:
changed to containsAll